crypto/src/external_commit.rs

```diff
@@ -0,0 +1,243 @@
+
+                    // we merge the commit and update the local state
+                    central
+                        .merge_pending_group_from_external_commit(
+                            &conversation_id,
+                            MlsConversationConfiguration::default(),
+                        )
+                        .await
+                        .unwrap();
+
+                    assert_eq!(central.group(&conversation_id).unwrap().members().unwrap().len(), 2);
+                })
+            })
+            .await
+        }
+
+        #[apply(all_credential_types)]
+        #[wasm_bindgen_test]
+        pub async fn test_join_by_external_commit_bad_epoch(credential: CredentialSupplier) {
+            run_test_with_central(credential, move |mut central| {
+                Box::pin(async move {
+                    let central = &mut central[0];
+                    let conversation_id = b"conversation".to_vec();
+                    let (alice_backend, mut alice) = person("alice", credential).await;
+                    let (_, bob) = person("bob", credential).await;
+
+                    // create alice group
+                    let mut alice_group = MlsConversation::create(
+                        conversation_id.clone(),
+                        alice.local_client_mut(),
+                        MlsConversationConfiguration::default(),
+                        &alice_backend,
+                    )
+                    .await
+                    .unwrap();
+
+                    // export the group from alice
+                    let state = alice_group
+                        .group
+                        .export_public_group_state(&alice_backend)
+                        .await
+                        .unwrap();
+                    let pgs_encoded: Vec<u8> = state.tls_serialize_detached().expect("Error serializing PGS");
+
+                    let verifiable_state = VerifiablePublicGroupState::tls_deserialize(&mut pgs_encoded.as_slice())
+                        .expect("Error deserializing PGS");
+
+                    // try to make and external join into alice's group
+                    let (_, message) = central.join_by_external_commit(verifiable_state).await.unwrap();
+
+                    // alice adds a new member to the group before receiving an ack from the
+                    // external join
+                    alice_group.add_members(&mut [bob], &alice_backend).await.unwrap();
+
+                    // receive the ack from the external join with outdated epoch
+                    // should fail because of the wrong epoch
+                    alice_group
+                        .decrypt_message(&message.to_bytes().unwrap(), &alice_backend)
+                        .await
+                        .unwrap_err();
+                    assert_eq!(alice_group.members().unwrap().len(), 2);
+
+                    // lets try again
```

Comment:
isn't the rest of the test redundant with `test_join_by_external_commit` ?
You test method just wants to check what happens if you receive outdated external commit